src/app/shared/constants/flags/spell_dbc_item_flags.ts

```diff
@@ -0,0 +1,193 @@
+import { Flag } from '../../types/general';
+
+export const SPELL_DBC_INVENTORY_TYPE: Flag[] = [
+  { bit: 0, name: 'HEAD' },
+  { bit: 1, name: 'NECK' },
+  { bit: 2, name: 'SHOULDERS' },
+  { bit: 3, name: 'BODY' },
+  { bit: 4, name: 'CHEST' },
+  { bit: 5, name: 'WAIST' },
+  { bit: 6, name: 'LEGS' },
+  { bit: 7, name: 'FEET' },
+  { bit: 8, name: 'WRISTS' },
+  { bit: 9, name: 'HANDS' },
+  { bit: 10, name: 'FINGER' },
+  { bit: 11, name: 'TRINKET' },
+  { bit: 12, name: 'WEAPON' },
+  { bit: 13, name: 'SHIELD' },
+  { bit: 14, name: 'RANGED' },
+  { bit: 15, name: 'CLOAK' },
+  { bit: 16, name: '2HWEAPON' },
+  { bit: 17, name: 'BAG' },
+  { bit: 18, name: 'TABARD' },
+  { bit: 19, name: 'ROBE' },
+  { bit: 20, name: 'WEAPONMAINHAND' },
+  { bit: 21, name: 'WEAPONOFFHAND' },
+  { bit: 22, name: 'HOLDABL' },
+  { bit: 23, name: 'AMMO' },
+  { bit: 24, name: 'THROWN' },
+  { bit: 25, name: 'RANGEDRIGHT' },
+  { bit: 26, name: 'QUIVER' },
+  { bit: 27, name: 'RELIC' },
```

Comment:
IIRC we already have these names saved in the project even the others below, you could try to re-use them instead of adding this redundancy

Reply:
check `item-class.ts`

Reply:
`item-class.ts` is an Option type, the DBC uses flags. I didn't found a simple solution to convert Option<->Flag.

Reply:
you can write it as:
```typescript
export const ITEM_CLASS_FLAGS: Flag[] = ITEM_CLASS.map(option => ({ bit: option.value as number, name: option.name }));
```

for `SPELL_DBC_INVENTORY_TYPE` you could use `INVENTORY_TYPE` but keep in mind that inventory_type starts from `'NON_EQUIP'` and value 0, while spell_dbc_inventory_type starts with `1` as you wrote above.

Then you need to skip the first value in this way:
```typescript
export const SPELL_DBC_INVENTORY_TYPE: Flag[] = INVENTORY_TYPE.slice(1, INVENTORY_TYPE.length).map(option => ({ bit: (option.value as number) - 1, name: option.name }));
```

Reply:
if you are not familiar with these tricks of javascript I could do it after that you finish the PR or at least this part.

Reply:
I changed the code and reused the existing options where possible.